mlir/include/mlir/Dialect/LLVMIR/GENXTypes.h

```diff
@@ -0,0 +1,72 @@
+//===- GENXTypes.h - MLIR GENX Types ----------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares the types in the GENX dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_DIALECT_LLVMIR_GENXTYPES_H_
+#define MLIR_DIALECT_LLVMIR_GENXTYPES_H_
+
+#include "mlir/Dialect/LLVMIR/GENXEnums.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/Diagnostics.h"
+#include "mlir/IR/Location.h"
+#include "mlir/IR/TypeSupport.h"
+#include "mlir/IR/Types.h"
+
+namespace mlir {
+namespace GENX {
+
+namespace detail {
+struct JointMatrixTypeStorage;
+} // namespace detail
+
+class GENXType : public Type {
+public:
+  using Type::Type;
+
+  static bool classof(Type type);
+
+  std::optional<int64_t> getSizeInBytes() const;
+};
+
+class CompositeType : public GENXType {
```

Comment:
We only have one custom type (JointMatrixType). We could make that type inherit directly from GENXType if we want to and avoid creating the intermediate CompositeType. What do you guys think?

Reply:
I am inclined to remove the composite type class.

Reply:
removed